contrib/migrate/migrate.c

```diff
@@ -0,0 +1,841 @@
+  	[ds].[db_name], \
+  	[ds].[user_name], \
+  	[ds].[properties], \
+  	[ds].[owner].[name], \
+  	[ds].[comment] \
+  from \
+  	[_db_server] [ds] \
+  where \
+  	current_user = 'DBA' \
+    or { [ds].[owner].[name] } subseteq ( \
+      select set {current_user} + coalesce (sum (set {[t].[g].[name]}), set { }) \
+      from [db_user] [u], table([groups]) as [t]([g]) \
+      where [u].[name] = current_user \
+    ) \
+    or { [ds] } subseteq ( \
+      select sum (set {[au].[class_of]}) \
+      from [_db_auth] [au] \
+      where {[au].[grantee].[name]} subseteq ( \
+                select set {current_user} + coalesce (sum (set {[t].[g].[name]}), set { }) \
+                from [db_user] [u], table([groups]) as [t]([g]) \
+                where [u].[name] = current_user \
+              ) \
+            and [au].[auth_type] = 'SELECT' \
+    )",
+
+  "CREATE TABLE [_db_synonym] ( \
+  	[unique_name] CHARACTER VARYING(255) NOT NULL, \
+  	[name] CHARACTER VARYING(255) NOT NULL, \
+  	[owner] [db_user] NOT NULL, \
+  	[is_public] INTEGER DEFAULT 0 NOT NULL, \
+  	[target_unique_name] CHARACTER VARYING(255) NOT NULL, \
+  	[target_name] CHARACTER VARYING(255) NOT NULL, \
+  	[target_owner] [db_user] NOT NULL, [comment] CHARACTER VARYING(2048), \
+  	CONSTRAINT [pk__db_synonym_unique_name] PRIMARY KEY  ([unique_name]), \
+  	INDEX [i__db_synonym_name_owner_is_public] ([name], [owner], [is_public]) \
+  ) DONT_REUSE_OID, COLLATE utf8_bin",
+
+  "create view [db_synonym] ( \
+  	[synonym_name] CHARACTER VARYING(255), \
+  	[synonym_owner_name] CHARACTER VARYING(255), \
+  	[is_public_synonym] CHARACTER VARYING(3), \
+  	[target_name] CHARACTER VARYING(255), \
+  	[target_owner_name] CHARACTER VARYING(255), \
+  	[comment] CHARACTER VARYING(2048) \
+  ) as \
+  SELECT \
+  	[s].[name] AS [synonym_name], \
+  	CAST ([s].[owner].[name] AS VARCHAR(255)) AS [synonym_owner_name], \
+  	CASE WHEN [s].[is_public] = 1 THEN 'YES' ELSE 'NO' END AS [is_public_synonym], \
+  	[s].[target_name] AS [target_name], \
+  	CAST ([s].[target_owner].[name] AS VARCHAR(255)) AS [target_owner_name], \
+  	[s].[comment] AS [comment] \
+	FROM \
+  [_db_synonym] [s] \
+  WHERE \
+    CURRENT_USER = 'DBA' \
+    OR [s].[is_public] = 1 \
+    OR ( \
+      [s].[is_public] = 0 \
+      AND { [s].[owner].[name] } SUBSETEQ ( \
+              SELECT SET {CURRENT_USER} + COALESCE (SUM (SET {[t].[g].[name]}), SET { }) \
+              FROM [db_user] [u], TABLE([groups]) AS [t]([g]) \
+              WHERE [u].[name] = CURRENT_USER \
+             ) \
+    )",
+
+  /* set system class for newly added tables and views */
+  "update _db_class set is_system_class = 1 where class_name in ('_db_server', 'db_server', '_db_synonym', 'db_synonym')"
+};
+
+static char *rename_query = "select \
+  	'rename table ' || class_name || ' to [' || lower (owner.name) || '.' || class_name || '] ' as q \
+   from  _db_class \
+   where is_system_class % 8 = 0";
+
+static char *update_unique_name =
```

Comment:
I think the below 2 variable names need consistency.

```
/* Update class_name and unique_name except for system classes. */
static char *update_unique_name = ...

/* Only system classes update unique_name. */
static char *update_db_class = ...
```

Reply:
Thanks, fixed as "update_db_class_not_for_system_classes" and "update_db_class_for_system_classes"